ivi-layermanagement-api/ilmInput/src/ilm_input.c

```diff
@@ -333,3 +333,27 @@ ilm_getInputFocus(t_ilm_surface **surfaceIDs, ilmInputDevice **bitmasks,
     return ILM_SUCCESS;
 
 }
+
+ILM_EXPORT ilmErrorTypes
+ilm_getDefaultSeat(t_ilm_string *seat_name)
+{
+    struct ilm_control_context *ctx = NULL;
+    struct seat_context *seat = NULL;
+
+    if (seat_name == NULL) {
+        fprintf(stderr, "Invalid Argument\n");
+        return ILM_FAILED;
+    }
+
+    ctx = sync_and_acquire_instance();
+    wl_list_for_each(seat, &ctx->wl.list_seat, link) {
+        if (seat->is_default) {
+            *seat_name = strdup(seat->seat_name);
```

Comment:
@khangtb1 please check if udev defines the max length of the seat name and use this number in the strndup function:
udev: https://github.com/systemd/systemd/tree/main/src/udev

Reply:
I looked to libinput, libudev and linsystemd, I see that:

1. Weston calls libinput_seat_get_logical_name() from libinput to get seatname.
2. libinput calls udev_device_get_property_value() from libudev to get value of key="WL_SEAT" 
3. libudev calls sd_device_get_property_value() from libsystemd to get value of key "WL_SEAT" 
4. The keys and values are defines in udev file, need to read the file to parse data. There is a SIZE_MAX when reading virtual file: [systemd/fileio.h at main · systemd/systemd · GitHub](https://github.com/systemd/systemd/blob/main/src/basic/fileio.h#L89) 
5. Don't see the max size of value: [systemd/sd-device.c at main · systemd/systemd · GitHub](https://github.com/systemd/systemd/blob/main/src/libsystemd/sd-device/sd-device.c#L111)

=> seat name in udev maybe too long, but it can't bigger than SIZE_MAX. So, here I'd like to check the valid of pointer instead of using strndup. what do you think?